BUILD_SHARED_LIBS=ON  QJS_BUILD_LIBC=ON fails to build

Version: 0.10.1
OS: debian/trixie/sid
Building libqjs.so works fine with
cmake -B build -DBUILD_SHARED_LIBS=ON

However, it fails with
cmake -B build -DBUILD_SHARED_LIBS=ON -DQJS_BUILD_LIBC=ON

with the following result:
```
LANG=C cmake --build build
[ 36%] Built target qjs
[ 42%] Linking C executable qjsc
/usr/bin/ld: CMakeFiles/qjsc.dir/qjsc.c.o: in function `output_object_code':
./build/./qjsc.c:170:(.text+0x3c2): undefined reference to `js_std_dump_error'
/usr/bin/ld: CMakeFiles/qjsc.dir/qjsc.c.o: in function `jsc_module_loader':
./build/./qjsc.c:242:(.text+0x42d): undefined reference to `js_load_file'
/usr/bin/ld: CMakeFiles/qjsc.dir/qjsc.c.o: in function `compile_file':
./build/./qjsc.c:280:(.text.startup+0x7e6): undefined reference to `js_load_file'
/usr/bin/ld: ./build/./qjsc.c:280:(.text.startup+0x8bf): undefined reference to `js_load_file'
/usr/bin/ld: ./build/./qjsc.c:280:(.text.startup+0x9ab): undefined reference to `js_load_file'
/usr/bin/ld: ./build/./qjsc.c:296:(.text.startup+0xf8d): undefined reference to `js_std_dump_error'
collect2: error: ld returned 1 exit status
gmake[2]: *** [CMakeFiles/qjsc.dir/build.make:102: qjsc] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:135: CMakeFiles/qjsc.dir/all] Error 2
```

If I disable completely the qjsc target, I get similar errors for qjs_exe target.

--- a/quickjs-libc.h
+++ b/quickjs-libc.h
@@ -34,26 +34,40 @@
 extern "C" {
 #endif
 
-JSModuleDef *js_init_module_std(JSContext *ctx, const char *module_name);
-JSModuleDef *js_init_module_os(JSContext *ctx, const char *module_name);
-JSModuleDef *js_init_module_bjson(JSContext *ctx, const char *module_name);
-void js_std_add_helpers(JSContext *ctx, int argc, char **argv);
-int js_std_loop(JSContext *ctx);
-JSValue js_std_await(JSContext *ctx, JSValue obj);
-void js_std_init_handlers(JSRuntime *rt);
-void js_std_free_handlers(JSRuntime *rt);
-void js_std_dump_error(JSContext *ctx);
-uint8_t *js_load_file(JSContext *ctx, size_t *pbuf_len, const char *filename);
-int js_module_set_import_meta(JSContext *ctx, JSValueConst func_val,
-                              bool use_realpath, bool is_main);
-JSModuleDef *js_module_loader(JSContext *ctx,
-                              const char *module_name, void *opaque);
-void js_std_eval_binary(JSContext *ctx, const uint8_t *buf, size_t buf_len,
-                        int flags);
-void js_std_promise_rejection_tracker(JSContext *ctx, JSValueConst promise,
-                                      JSValueConst reason,
-                                      bool is_handled, void *opaque);
-void js_std_set_worker_new_context_func(JSContext *(*func)(JSRuntime *rt));
+#if defined(__GNUC__) || defined(__clang__)
+#define JS_EXTERN __attribute__((visibility("default")))
+#else
+#define JS_EXTERN /* nothing */
+#endif
+
+JS_EXTERN JSModuleDef *js_init_module_std(JSContext *ctx,
+                                          const char *module_name);
+JS_EXTERN JSModuleDef *js_init_module_os(JSContext *ctx,
+                                         const char *module_name);
+JS_EXTERN JSModuleDef *js_init_module_bjson(JSContext *ctx,
+                                            const char *module_name);
+JS_EXTERN void js_std_add_helpers(JSContext *ctx, int argc, char **argv);
+JS_EXTERN int js_std_loop(JSContext *ctx);
+JS_EXTERN JSValue js_std_await(JSContext *ctx, JSValue obj);
+JS_EXTERN void js_std_init_handlers(JSRuntime *rt);
+JS_EXTERN void js_std_free_handlers(JSRuntime *rt);
+JS_EXTERN void js_std_dump_error(JSContext *ctx);
+JS_EXTERN uint8_t *js_load_file(JSContext *ctx, size_t *pbuf_len,
+                                const char *filename);
+JS_EXTERN int js_module_set_import_meta(JSContext *ctx, JSValueConst func_val,
+                                        bool use_realpath, bool is_main);
+JS_EXTERN JSModuleDef *js_module_loader(JSContext *ctx,
+                                        const char *module_name, void *opaque);
+JS_EXTERN void js_std_eval_binary(JSContext *ctx, const uint8_t *buf,
+                                  size_t buf_len, int flags);
+JS_EXTERN void js_std_promise_rejection_tracker(JSContext *ctx,
+                                                JSValueConst promise,
+                                                JSValueConst reason,
+                                                bool is_handled,
+                                                void *opaque);
+JS_EXTERN void js_std_set_worker_new_context_func(JSContext *(*func)(JSRuntime *rt));
+
+#undef JS_EXTERN
 
 #ifdef __cplusplus
 } /* extern "C" { */

